src/common/common.py

```diff
@@ -607,11 +972,17 @@ def get_dataframe_mem_useage(df):
     Returns:
         float: The memory usage of the DataFrame in megabytes.
     """
-    # Calculate the memory usage of the DataFrame in bytes
-    memory_usage_bytes = df.memory_usage(deep=True).sum()
-    # Convert bytes to megabytes
-    memory_usage_mb = memory_usage_bytes / (1024**2)
-    return memory_usage_mb
+    if not PANDAS_AVAILABLE:
```

Comment:
Pandas should be always available. It is just the plotting packages that are not fixed as dependencies so we should only check for them.